bssl-compat/source/RSA_decrypt.cc

```diff
@@ -0,0 +1,41 @@
+#include <openssl/rsa.h>
+#include <ossl.h>
+
+
+/*
+ * https://github.com/google/boringssl/blob/cd0b767492199a82c7e362d1a117e8c3fef6b943/include/openssl/rsa.h#L259
+ * https://www.openssl.org/docs/man3.0/man3/RSA_private_decrypt.html
+ */
+extern "C" int RSA_decrypt(RSA *rsa, size_t *out_len, uint8_t *out, size_t max_out, const uint8_t *in, size_t in_len, int padding) {
+  size_t min_out = RSA_size(rsa);
+
+  switch(padding) {
```

Comment:
It's not clear to me where the values below (11, 42) comes from.

Reply:
Those numbers come from the OpenSSL [man page for `RSA_private_decrypt()`](https://www.openssl.org/docs/man3.0/man3/RSA_private_decrypt.html) (the function being used to implement BoringSSL's `RSA_decrypt()`).
If the encrypted data includes padding, then the decrypted data does not necessarily need a buffer of `RSA_size(rsa)` bytes to be decrypted into. The 11 and 42 values correspond to the size of the padding in the two different padding schemes. However, having read the BoringSSL [documentation for `RSA_decrypt()`](https://github.com/google/boringssl/blob/098695591f3a2665fccef83a3732ecfc99acdcdd/src/include/openssl/rsa.h#L251) again, it states that _"The `max_out` argument must be, at least, `RSA_size` in order to ensure success."_. Therefore, to be as faithful as possible to the BoringSSL documentation, I will modify/simplify the code to match. Thanks for pointing this out.

Reply:
Thanks for the detailed explanation.